File: python-algorithm/ds_3_hash/L242.py
# -*- coding: utf-8 -*-


class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        d = {}
        for c in s:
            if c not in d:
                d[c]  = 1
            else:
                d[c] += 1
        for c in t:
            if c not in d:
                return False
            else:
                d[c] -= 1
        for k,v in d.items():
            if v != 0:
                return False
        return True

File: ds_3_hash/L242.cpp

#include <iostream>
#include <vector>
using namespace std;
class Solution {
public:
    bool isAnagram(string s, string t) {
        vector<int> record(26,0);
        for(int i=0; i<s.size(); ++i)
            record[s[i]-'a']++;
        for(int i=0; i<t.size(); ++i)
            record[t[i]-'a']--;
        for(int i=0; i< record.size(); ++i){
            if(record[i] != 0)
                return false;
        }
        return true;
    }
};
// 能用数组代替表达的时候选数组，效率高过哈希表
// 还有一种方法比较22个字母排序完的结果是否一样